flask_server.py

```diff
@@ -234,98 +168,144 @@ def uploadData(dataset_name=None, headerfile=None, zipfile=None, project_name=No
-        is_test=False):
-    """Save uploaded time series data files and begin featurization.
-
-    Handles POST form submission.
 
-    Returns
-    -------
-    Redirects to featurizationPage, see that function for output
-    details.
+        return to_json({"status": "success"})
+    elif request.method == "GET":
+        if dataset_id is not None:
+            dataset_info = m.Dataset.get(m.Dataset.id == dataset_id)
+        else:
+            dataset_info = [d for p in m.Project.all(USERNAME)
+                            for d in p.datasets]
+
+        return to_json(
+            {
+                "status": "success",
+                "data": dataset_info
+            })
+    elif request.method == "DELETE":
+        if dataset_id is None:
+            return to_json(
+                {
+                    "status": "error",
+                    "message": "Invalid request - data set ID not provided."
+                })
+        d = m.Dataset.get(m.Dataset.id == dataset_id)
+        if d.is_owned_by(USERNAME):
+            d.delete_instance()
+        else:
+            raise UnauthorizedAccess("User not authorized for project.")
+
+        return to_json({"status": "success"})
+    elif request.method == "PUT":
+        if dataset_id is None:
+            return to_json(
+                {
+                    "status": "error",
+                    "message": "Invalid request - data set ID not provided."
+                })
+        # TODO!
+        return to_json(
+            {
+                "status": "error",
+                "message": "Functionality for this endpoint is not yet implemented."
+            })
 
+
+@app.route('/features', methods=['POST', 'GET'])
+@app.route('/features/<featureset_id>', methods=['GET', 'PUT', 'DELETE'])
+def Features(featureset_id=None):
+    """
     """
     # TODO: ADD MORE ROBUST EXCEPTION HANDLING (HERE AND ALL OTHER FUNCTIONS)
     if request.method == 'POST':
-        post_method = "browser"
         # Parse form fields
-        featureset_name = str(request.form["featureset_name"]).strip()
-        if featureset_name == "":
-            return jsonify({
-                "message": ("Feature Set Title must contain non-whitespace "
-                            "characters. Please try a different title."),
-                "type": "error"})
-        dataset_id = str(request.form["featureset_dataset_select"]).strip()
-        project_name = (str(request.form["featureset_project_name_select"]).
-                        strip().split(" (created")[0])
-        features_to_use = request.form.getlist("features_selected")
-        custom_script_tested = str(request.form["custom_script_tested"])
-        if custom_script_tested == "yes":
-            custom_script = request.files["custom_feat_script_file"]
+        featureset_name = request.form["Feature Set Title"].strip()
+        dataset_id = request.form["Select Dataset"].strip()
+        proj_key = request.form["Select Project"].strip()
```

Comment:
I think we are sticking to `*_id` instead of `*_key` since all the new tables should have an `id` column.